api/Dockerfile

```diff
@@ -26,23 +26,25 @@ ENV PYTHONUNBUFFERED=1
 ENV PIP_NO_CACHE_DIR=1
 ENV PIP_NO_COLOR=1
 
-# Activate the virtualenv
-ENV PATH="/venv/bin:$PATH"
-
 # - Install system packages needed for building Python dependencies
 # - Create a virtualenv inside `/venv`
-# - Install Pipenv to install Python dependencies
+# - Install PDM to install Python dependencies
 RUN apt-get update \
   && apt-get install -y python3-dev git \
   && rm -rf /var/lib/apt/lists/* \
-  && python -m venv /venv \
-  && pip install --upgrade pipenv
+  && pip install pdm~=2.14
```

Comment:
Is there a reason the version should be manually specified here? `2.15.0` is already out for example.

Reply:
Only reason is select a specific version is for lockfile compataibility stuff with PDM, similar to pnpm. FWIW `~=` will resolve to 2.15.0 I think? But if we're going to use a specific version of PDM, we should document and parameterise it like we do for pnpm :thinking: Ideally we're just using the latest PDM though, however "unreproducible" that technically could be... If we're very worried about it, we should use `https://tracker.debian.org/pkg/pdm` instead, but we didn't pin pipenv before, so I think it might not be too necessary for pdm either.

Reply:
`~=2.14` means `>=2.14,==2.*` so it gives us both the benefit of any new minor updates and patches that come out but prevents updating to a v3 that might break compatibility. Python's `pyproject.toml` doesn't have an equivalent of Node's `engines` so I don't think making this a parameter will work.

Reply:
It could be encoded in the `pyproject.toml` at the monorepo root, as just a comment (like we've done for the catalog's Python version), but I'm not very worried at all about this.